metashare/api/gh.py

```diff
@@ -138,3 +139,36 @@ def try_to_make_branch(repository, *, new_branch, base_branch):
                 counter += 1
             else:
                 raise
+
+
+def validate_gh_hook_signature(
+    *, hook_secret: bytes, signature: bytes, message: bytes
+) -> bool:
+    local_signature = "sha1=" + hmac.new(hook_secret, message, "sha1").hexdigest()
+    # print("\033[1;92m======>", local_signature, "\033[0m")
+    return hmac.compare_digest(local_signature, signature)
+
+
+def normalize_commit(commit):
+    """
+    This takes commits either in the JSON format provided by a GitHub
+    webhook, or the object format provided by github3.py, and returns a
+    normalized Python dict.
+    """
+    # TODO: make this right.
+    if isinstance(commit, dict):
+        return commit
```

Comment:
@wlonk I think you can go ahead an implement this?

Reply:
Oh yes, I just forgot to because it's close enough that the tests passed.